osu.Game/Database/ILive.cs

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+
+namespace osu.Game.Database
+{
+    /// <summary>
+    /// A wrapper to provide access to database backed classes in a thread-safe manner.
+    /// </summary>
+    /// <typeparam name="T">The databased type.</typeparam>
+    public interface ILive<out T> where T : class
+    {
+        Guid ID { get; }
```

Comment:
Shame this can't be shared with `IHasGuidPrimaryKey` but that one also has a setter which is undesirable here, so...

Reply:
Not actually sure the setter is required in that interface, will take a look.

Reply:
The setter can be dropped (have done so) but we can't use the interface here until EF is dropped (due to the primary key type differing). I've made a TODO so we don't forget.